src/Schema/Builder.php

```diff
@@ -64,6 +79,24 @@ public function getIndexListing($table)
         return $processor->processIndexListing($results);
     }
 
+    /**
+     * @param string $table
+     * @return string[]
+     */
+    public function getForeignListing($table)
```

Comment:
Maybe you can use [getForeignKeys()](https://github.com/laravel/framework/blob/eb948833d513088249d73775586d8fe0ccf48158/src/Illuminate/Database/Schema/Builder.php#L376) instead?

Reply:
I feel like in this file its more in line with the other method `getIndexListing()` just above?

Reply:
But happy to change that too

Reply:
It also uses the `$processor->processIndexListing()` method internally, its basically a 1:1 copy of `getIndexListing()` so I thought I would keep the naming convention the same

Reply:
The internals of `getForeignKeys()` looks like the below, so maybe we can just implement `processForeignKeys` in the PostProcessor. That way it's more compatible with other databases.
```
$table = $this->connection->getTablePrefix().$table;

return $this->connection->getPostProcessor()->processForeignKeys(
    $this->connection->selectFromWriteConnection($this->grammar->compileForeignKeys($table))
);
```

Reply:
If we want to be consistent with (https://github.com/laravel/framework/blob/eb948833d513088249d73775586d8fe0ccf48158/src/Illuminate/Database/Schema/Builder.php)

shouldnt `getIndexListing` be `getIndexes` too? And then the Processor should have `processIndexes` and `processForeignKeys`?

Reply:
Regardless I changed getForeignKeyListing to getForeignKeys and added processForeignKeys, this PR should be ready to merge

Reply:
FYI `getIndexes` was [added to the framework last week](https://github.com/laravel/framework/pull/49204).

Reply:
Ah makes sense